src/back/server.py

```diff
@@ -6,171 +6,133 @@
-            img = cv2.warpAffine(img, M, (cols, rows))
-
-            # rebuild a VideoFrame, preserving timing information
-            new_frame = VideoFrame.from_ndarray(img, format="bgr24")
-            new_frame.pts = frame.pts
-            new_frame.time_base = frame.time_base
-            return new_frame
-        elif self.transform == "skeleton":
-            # Convert the aiortc frame to an array
-            img = frame.to_ndarray(format="bgr24")
-
-            # Process the frame for skeleton
-            img = await process_frame_for_skeleton(img)
-
-            # Rebuild a VideoFrame, preserving timing information
-            new_frame = VideoFrame.from_ndarray(img, format="bgr24")
-            new_frame.pts = frame.pts
-            new_frame.time_base = frame.time_base
-            return new_frame
-        else:
-            return frame
+source_video = VideoStreamTrack()
+NO_TRANSFORM: str = "none"
+__transformed_tracks = dict()
+
+
+def apply_transform(track: MediaStreamTrack, transform: str) -> MediaStreamTrack:
+    if transform == NO_TRANSFORM:
+        return track
+    if transform not in __transformed_tracks:
+        __transformed_tracks[transform] = VideoTransformTrack(track, transform)
+    return __transformed_tracks[transform]
+
+
+consumer_schema = {
+    "type": "object",
+    "properties": {
+        "sdp": {"type": "string"},
+        "type": {"type": "string"},
+        "video_transform": {"type": "string"},
+    },
+    "required": ["sdp", "type", "video_transform"],
+}
+
+broadcast_schema = {
+    "type": "object",
+    "properties": {
+        "sdp": {"type": "string"},
+        "type": {"type": "string"},
+    },
+    "required": ["sdp", "type"],
+}
+
 
 async def consumer(request):
     if request.method == "OPTIONS":
         return web.Response(
             content_type="application/json",
-            headers={"Access-Control-Allow-Origin": "*", 
-                        "Access-Control-Allow-Credentials": "true", 
-                        "Access-Control-Allow-Methods": "POST, GET, OPTIONS", 
-                        "Access-Control-Allow-Headers": "Content-Type"},
+            headers={
+                "Access-Control-Allow-Origin": "*",
+                "Access-Control-Allow-Credentials": "true",
+                "Access-Control-Allow-Methods": "POST, GET, OPTIONS",
+                "Access-Control-Allow-Headers": "Content-Type"
+            },
         )
 
-    params = await request.json()
-    description = RTCSessionDescription(sdp=params["sdp"], type=params["type"])
+    body = await request.json()
+    try:
+        validate(instance=body, schema=consumer_schema)
+    except ValidationError as error:
+        raise web.HTTPBadRequest(reason=error.message)
 
+    description = RTCSessionDescription(sdp=body["sdp"], type=body["type"])
     pc = RTCPeerConnection()
     pc_id = "PeerConnection(%s)" % uuid.uuid4()
-    pcs.add(pc)
+    peer_connections.add(pc)
 
     def log_info(msg, *args):
```

Comment:
please don't interleave function definition with other code logic

Reply:
Wont fix: this is a closure for more convenient logging. And this is from the aiortc example code.